### DISCUSSION TITLE
Rotate screen and touch screen input when using framebuffer (StartLinuxFbDev)

### DISCUSSION THREAD

**Discussion by rickvanderheijden** [AvaloniaUI/Avalonia#14326] [Q&A]
Hi all,

I am creating an application using a Linux based embedded device. The manufacturer of the device indicated that in one of the next releases, they will drop the X Windows System (X11) support with the next kernel update. To see if we can overcome this, I started experimenting with the framebuffer option offered in Avalonia. 

Because the physical screen in the device is rotated 90 degrees, the startup script instructs X11 to rotate the display and swap the touch input axes. The X11 based Avalonia application works like a charm when doing so. However, this seems not to be possible for the framebuffer option. Rotating the application is possible by setting the RotateTransform property, The image below shows how the reproducer app is shown on the screen when it is rotated. I cannot get the touch input to rotate. The red rectangle indicates the area where I need to press to activate the button. Coordinates also look a bit off to me. 

I was expecting that adding a CalibrationMatrix to the EvDevBackend could somehow fix this, but did not have any success yet.

It would be great to hear from someone who has any experience with this. 

Thanks,
Rick

![image](https://github.com/AvaloniaUI/Avalonia/assets/31483005/fd4dc206-b9cd-4156-81f1-15e9855f5214)

**Comment by theredcmdcraft** [comment]
did you have a running application with avalonia running in framebuffer? how did you do that? and for your problem: red rectangle is mirrored not rotated? i think the rectangle is wrong?
